### ISSUE TITLE
SMTP protocol is matched case-sensitively

### ISSUE THREAD

**Issue by fatalbanana** [linyows/warp#35]
Protocol is matched case-sensitively where-as the receiving MTA probably doesn't care. This can cause metadata not to be collected or false protocol synchronization errors to be generated.

Also: elements like `MAIL FROM` might/not contain one/more spaces before `<`.

**Comment by linyows** [comment]
Oh, Thanks!

**Comment by linyows** [comment]
Thank you for the valuable information. Do you have plans to use Warp for any specific purpose? If you have any use cases, sharing them would be motivating for development, so I would be happy to hear about them.

**Comment by fatalbanana** [comment]
> Thank you for the valuable information. Do you have plans to use Warp for any specific purpose?

Yes, I have a use-case in mind but it's not yet clear to me that I'll be pursuing it. The idea is to implement support for scanning messages with [Rspamd](https://rspamd.com/)  - and inserting fake rejections into the stream if messages would be considered bad.

**Comment by linyows** [comment]
That's a good idea. I wonder if it would be good to have an extension like milter in Warp for that?

**Comment by fatalbanana** [comment]
> That's a good idea. I wonder if it would be good to have an extension like milter in Warp for that?

Milter support could be generally useful but Rspamd's [HTTP protocol](https://rspamd.com/doc/architecture/protocol.html) would be a better target for an Rspamd integration (in that it's relatively flexible vs milter or other integration).
